src/kernel_module.cpp

```diff
@@ -135,8 +135,12 @@ Value KernelModule::exit_bang(Env *env, Value status) {
 
 Value KernelModule::Integer(Env *env, Value value, Value base, Value exception) {
     nat_int_t base_int = 0; // default to zero if unset
-    if (base && base->is_integer())
+    if (base) {
+        if (!base->is_integer() && base->respond_to(env, "to_int"_s))
+            base = base->send(env, "to_int"_s);
+
```

Comment:
I think we need an assert_type() call here in case to_int returns something else.

Reply:
Well, actually...

I just tried this in IRB:
```
irb(main):001:0> Integer('123', Object.new.tap { |x| x.define_singleton_method(:
to_int) { 8 } })
=> 83
irb(main):002:0> Integer('123', '8')
=> 123
```
(Keep in mind: `String` does not have a `to_int` method, it has a `to_i` but that is not what's in use here). So if the base argument does not have a `to_int` method, it gets ignored. I'm not sure whether this is the intended behaviour, given that `Kernel.Integer` is otherwise more strict than the `to_i` methods.

I already added a spec update upstream for this issue, that hasn't been merged yet, so I'm just going out on a limb and drop a question here. We might want to wait for a second before we merge this change.

Reply:
>We might want to wait for a second before we merge this change.

OK, holding. 😄